transformations/sentence_additions/transformation.py

```diff
@@ -0,0 +1,39 @@
+from interfaces.SentenceOperation import SentenceOperation
+from tasks.TaskTypes import TaskType
+from transformers import pipeline, set_seed
+
+"""
+Adds generated sentence into provided sentences or paragraph to create adversarial examples.
+"""
+
+
+class SentenceAdditions(SentenceOperation):
+    tasks = [
+        TaskType.TEXT_CLASSIFICATION,
+        TaskType.TEXT_TO_TEXT_GENERATION,
+    ]
+    languages = ["en"]
+    heavy = True
+
+    def __init__(self, seed=0, max_outputs=1, max_length=75, model='gpt2-xl'):
+        super().__init__(seed, max_outputs=max_outputs)
+        self.seed = seed
+        self.model = model
+        self.max_outputs = max_outputs
+        self.max_length = max_length
+
+    def generate(self, sentence: str):
+        perturbed = self.sentence_additions(text=sentence)
+        return perturbed
+
+    def sentence_additions(self, text):
+        set_seed(self.seed)
+        generator = pipeline('text-generation', model=self.model)
+        outputs = generator(text, max_length = self.max_length, num_return_sequences = self.max_outputs)
```

Comment:
Okay, a couple of things here: Instead of passing direct text, do you think using prompt templates can force the model to generate paraphrases? (like "Paraphrase: ")

Reply:
Hi @kaustubhdhole,

Sorry I missed your message. That's a very interesting idea! Are you thinking about using prompt templates as inputs to generate paraphrases of the original text? Using a manual or automated approach to create prompt templates? 

I am new to prompt templates so I'll need to look into this further. I came across this article that covers some aspects of prompt engineering:

http://pretrain.nlpedia.ai/data/pdf/template.pdf

Feel free to share any resources that you feel may be useful for me to know as well. :)

Reply:
No problem! There are some which are interesting which have been studied in the context of GPT-3: https://arxiv.org/pdf/2102.09690.pdf and https://aclanthology.org/2020.emnlp-main.346.pdf

My question was if instead of simply passing the original text, you could test out by appending words like "Sentence: Paraphrase: ", etc. This could be effective later when people decide to replace the model from GPT-2 to say, GPT-3.

Reply:
No problem! You can check these two: https://arxiv.org/pdf/2102.09690.pdf, https://aclanthology.org/2020.emnlp-main.346.pdf. What I meant was that you could append your input with a piece of text (prompt) like "PARAPHRASE: " and try checking how it performs on that. Since few-shot prompting has shown impressive results in GPT-3, later users could pass GPT-3 or another model proven well on prompting. You could probably have a default prompt text defined in the constructor itself.

Reply:
Hi @kaustubhdhole ,

Thanks for sharing those resources! I really appreciate it :D

I'll add a default prompt text ("PARAPHASE: " as per your suggestion) in the constructor which the user can change if they want and also an option to turn the prompting on/off.